src/cpp/ext/otel/otel_server_call_tracer.h

```diff
@@ -21,28 +21,130 @@
 
 #include <grpc/support/port_platform.h>
 
+#include "absl/strings/strip.h"
+
 #include "src/core/lib/channel/call_tracer.h"
 #include "src/core/lib/channel/channel_args.h"
-#include "src/core/lib/resource_quota/arena.h"
 #include "src/cpp/ext/otel/otel_plugin.h"
 
 namespace grpc {
 namespace internal {
 
-class OpenTelemetryServerCallTracerFactory
-    : public grpc_core::ServerCallTracerFactory {
+// OpenTelemetryServerCallTracer implementation
+
+class OpenTelemetryServerCallTracer : public grpc_core::ServerCallTracer {
  public:
-  explicit OpenTelemetryServerCallTracerFactory(
-      OpenTelemetryPlugin* otel_plugin);
+  OpenTelemetryServerCallTracer(
+      OpenTelemetryPlugin* otel_plugin,
+      std::shared_ptr<OpenTelemetryPlugin::ScopeConfig> scope_config)
+      : start_time_(absl::Now()),
+        injected_labels_from_plugin_options_(
+            otel_plugin->plugin_options().size()),
+        otel_plugin_(otel_plugin),
+        scope_config_(std::move(scope_config)) {}
+
+  std::string TraceId() override {
+    // Not implemented
+    return "";
+  }
+
+  std::string SpanId() override {
+    // Not implemented
+    return "";
+  }
+
+  bool IsSampled() override {
+    // Not implemented
+    return false;
+  }
+
+  // Please refer to `grpc_transport_stream_op_batch_payload` for details on
+  // arguments.
+  void RecordSendInitialMetadata(
```

Comment:
This method is long enough that it should probably move to the .cc file.

https://google.github.io/styleguide/cppguide.html#Inline_Functions